examples/platform/esp32/common/TimeSync.h

```diff
@@ -0,0 +1,29 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+#if CONFIG_ENABLE_SNTP_TIME_SYNC
```

Comment:
Time sync seems specifically designed for time synchronization. Could we handle this in build rules (and maybe in the app itself) instead?

This pattern of including all headers all the time and ifdefing the entire content is not great.